docs/website/docs/general-usage/dataset-access/dataset.md

```diff
@@ -0,0 +1,238 @@
+# Get the first 50 items as a PyArrow table
+arrow_table = items_relation.limit(50).arrow()
+```
+
+#### Using `head()` to get the first 5 records
+
+```py
+df = items_relation.head().df()
+```
+
+### Select specific columns
+
+```py
+# Select only 'col1' and 'col2' columns
+items_list = items_relation.select("col1", "col2").fetchall()
+
+# Alternate notation with brackets
+items_list = items_relation[["col1", "col2"]].fetchall()
+
+# Only get one column
+items_list = items_relation["col1"].fetchall()
+
+```
+
+### Chain operations
+
+You can combine `select`, `limit`, and other methods.
+
+```py
+# Select columns and limit the number of records
+arrow_table = items_relation.select("col1", "col2").limit(50).arrow()
+```
+
+## Supported destinations
+
+All SQL and filesystem destinations supported by `dlt` can utilize this data access interface. For filesystem destinations, `dlt` [uses **DuckDB** under the hood](./sql-client.md#the-filesystem-sql-client) to create views from Parquet or JSONL files dynamically. This allows you to query data stored in files using the same interface as you would with SQL databases. If you plan on accessing data in buckets or the filesystem a lot this way, it is advised to load data as parquet instead of jsonl, as **DuckDB** is able to only load the parts of the data actually needed for the query to work.
+
+## Examples
+
+### Fetch one record as a tuple
+
+```py
+record = items_relation.fetchone()
+```
+
+### Fetch many records as tuples
+
+```py
+records = items_relation.fetchmany(chunk_size=10)
+```
+
+### Iterate over data with limit and column selection
+
+**Note:** When iterating over filesystem tables, the underlying DuckDB may give you a different chunk size depending on the size of the parquet files the table is based on.
+
+```py
+
+# Dataframes
+for df_chunk in items_relation.select("col1", "col2").limit(100).iter_df(chunk_size=20):
+    ...
+
+# Arrow tables
+for arrow_table in items_relation.select("col1", "col2").limit(100).iter_arrow(chunk_size=20):
+    ...
+
+# Python tuples
+for records in items_relation.select("col1", "col2").limit(100).iter_fetch(chunk_size=20):
+    # Process each modified DataFrame chunk
+    ...
+```
+
+## Advanced usage
+
+### Using custom SQL queries to create `ReadableRelations`
+
+You can use custom SQL queries directly on the dataset to create a `ReadableRelation`:
+
+```py
+# Join 'items' and 'other_items' tables
+custom_relation = dataset("SELECT * FROM items JOIN other_items ON items.id = other_items.id")
+arrow_table = custom_relation.arrow()
+```
+
+**Note:** When using custom SQL queries with `dataset()`, methods like `limit` and `select` won't work. Include any filtering or column selection directly in your SQL query.
+
+
+### Loading a `ReadableRelation` into a pipeline table
+
+Since the iter_arrow and iter_df methods are generators that iterate over the full ReadableRelation in chunks, you can use them as a resource for another (or even the same) dlt pipeline:
```

Comment:
```suggestion
Since the `iter_arrow` and `iter_df` methods are generators that iterate over the full `ReadableRelation` in chunks, you can use them as a resource for another (or even the same) `dlt` pipeline:
```